client-vms/tests/signature.test.ts

```diff
@@ -0,0 +1,144 @@
+  const tecdsaKeyName = "tecdsa_private_key";
+  const tecdsaDigestName = "tecdsa_digest_message";
+  const tecdsaSignatureName = "tecdsa_signature";
+  // Party names
+  const tecdsaKeyParty = "tecdsa_key_party";
+  const tecdsaDigestParty = "tecdsa_digest_message_party";
+  const tecdsaOutputParty = "tecdsa_output_party";
+
+  const privateKey: Uint8Array = secp256k1.utils.randomPrivateKey();
+  const publicKey: Uint8Array = secp256k1.getPublicKey(privateKey);
+
+  let digestMessage: Uint8Array;
+
+  let client: VmClient;
+
+  beforeAll(async () => {
+    const signer = await createSignerFromKey(PrivateKeyPerSuite.VmClient);
+    digestMessage = await hash("A deep message with a deep number: 42");
+
+    client = await new VmClientBuilder()
+      .seed("tests")
+      .bootnodeUrl(Env.bootnodeUrl)
+      .chainUrl(Env.nilChainUrl)
+      .signer(signer)
+      .build();
+  });
+
+  let privateKeyStoreId: Uuid;
+  it("store private key", async () => {
+    const permissions = ValuesPermissionsBuilder.default_owner(client.id)
+      .grantCompute(client.id, tecdsaProgramId)
+      .build();
+    privateKeyStoreId = await client
+      .storeValues()
+      .ttl(1)
+      .value(tecdsaKeyName, NadaValue.new_ecdsa_private_key(privateKey))
+      .permissions(permissions)
+      .build()
+      .invoke();
+    expect(privateKeyStoreId).toHaveLength(36);
+  });
+
+  it("retrieve private key", async () => {
+    const data = await client
+      .retrieveValues()
+      .id(privateKeyStoreId)
+      .build()
+      .invoke();
+
+    const values = data[tecdsaKeyName]!;
+    expect(values).toBeDefined();
+    expect(values.type).toBe("EcdsaPrivateKey");
+    expect(values.value).toEqual(privateKey);
+  });
+
+  let digestMessageStoreId: Uuid;
+  it("store digest message", async () => {
+    const permissions = ValuesPermissionsBuilder.default_owner(client.id)
+      .grantCompute(client.id, tecdsaProgramId)
+      .build();
+    digestMessageStoreId = await client
+      .storeValues()
+      .ttl(1)
+      .value(
+        tecdsaDigestName,
+        NadaValue.new_ecdsa_digest_message(digestMessage),
+      )
+      .permissions(permissions)
+      .build()
+      .invoke();
+    expect(digestMessageStoreId).toHaveLength(36);
+  });
+
+  it("retrieve digest message", async () => {
+    const data = await client
+      .retrieveValues()
+      .id(digestMessageStoreId)
+      .build()
+      .invoke();
+
+    const values = data[tecdsaDigestName]!;
+    expect(values).toBeDefined();
+    expect(values.type).toBe("EcdsaDigestMessage");
+    expect(values.value).toEqual(digestMessage);
+  });
+
+  let computeResultId: Uuid;
+  it("sign message", async () => {
+    computeResultId = await client
+      .invokeCompute()
+      .program(tecdsaProgramId)
+      .inputParty(tecdsaKeyParty, client.id)
+      .inputParty(tecdsaDigestParty, client.id)
+      .outputParty(tecdsaOutputParty, [client.id])
+      .valueIds(privateKeyStoreId, digestMessageStoreId)
```

Comment:
We should create a test for the computation using stored values. This wasn't tested and it was failing.